File: advent-of-code_2023/day1/part1.js
import {readFileSync} from "fs";

const data = readFileSync('input.txt', {encoding: "utf-8"});
const lines = data.split("\n");

const digits = lines.map(line => line.match(/\d/g));
console.log(digits.reduce((sum, digit) => {
    const num = digit.slice(0, 1) + digit.slice(-1);
    return sum + parseInt(num);
}, 0));

File: advent-of-code_2015/src/day2/part1.py
total = 0

with open("input.txt") as file:
    for data in file:
        numbers = list(map(int, data.split('x')))

        surface_area = (2 * numbers[0] * numbers[1]) + (2 * numbers[1] * numbers[2]) + (2 * numbers[0] * numbers[2])
        numbers.sort()
        extra = numbers[0] * numbers[1]

        total += surface_area + extra

print(total)
